.github/workflows/feature-serverless-framework-deployment.yml

```diff
@@ -53,8 +53,67 @@ jobs:
           path: build.tar
           retention-days: 1
 
-  e2e_aws:
-    name: AWS e2e test
+  unit_tests:
+    name: Unit tests
+    needs: [ build_client ]
+    runs-on: ubuntu-22.04
+    env:
+      working-directory: ./src
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Configure AWS credentials using EASE lab account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_KEY }}
+          aws-region: us-west-1
+
+      - name: Set up Go 1.21
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.21
+
+      - name: Download client artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: STeLLAR-build
+
+      - name: Untar client build
+        working-directory: ${{env.working-directory}}
+        run: tar --strip-components=1 -xvf ../build.tar -C .
+
+      - name: Unit tests setup
+        working-directory: ${{env.working-directory}}
+        run: go test -short -v ./setup/test/...
+
+      - name: Unit tests benchmarking
+        working-directory: ${{env.working-directory}}
+        run: go test -short -v ./benchmarking/...
+
+      - name: Unit tests util
+        working-directory: ${{env.working-directory}}
+        run: go test -short -v ./util/...
+
+      - name: Unit tests building
+        working-directory: ${{env.working-directory}}
+        run: go test -short -v ./setup/building/test/...
+
+      - name: Unit tests code-generation
+        working-directory: ${{env.working-directory}}
+        run: go test -short -v ./setup/code-generation/test/...
+
+      - name: Unit tests packaging
+        working-directory: ${{env.working-directory}}
+        run: go test -short -v ./setup/deployment/packaging/test/...
+
+      - name: Unit tests connection
+        working-directory: ${{env.working-directory}}
+        run: go test -short -v ./setup/deployment/connection/...
```

Comment:
please use `matrix` to avoid duplication and to run them in parallel
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs